happy_bank_core/logic/transaction.py

```diff
@@ -0,0 +1,46 @@
+from account import Account
+from account import accounts
+
+a1 = Account("#1234")
+a2 = Account("#12345")
+
+
+class Transaction:
+    @staticmethod
+    def check_balance(customer, amount):
+        """Verifies if customer has enough money to transfer"""
+        amount = float(amount)
+        balance = customer.deposit
+        if 0 < amount <= balance:
+            return True
+        else:
+            return False
+
+    @staticmethod
+    def transfer(sender, recipient, amount):
+        """Ensures transfer between 2 accounts"""
+        if sender.id == recipient.id:
+            print(
+                "Accounts are equal, please enter correct account ID\n"
+                "----------------------------------------------------"
+            )
+        elif not Transaction.check_balance(sender, amount):
+            print(
+                "insufficient funds or incorrect amount was entered, "
+                "please try again\n"
+                "----------------------------------------------------"
+            )
+        else:
```

Comment:
Could we add a custom exception that's raised when the condition is True? E.g. TransactionException with message property that inherits from general Exception. Then, we can have only 2 if conditions and get rid of else branch that returns a tuple of updated Account instances.

The print commands can be left here - will be replaced by logger call later.